From f56bd67f80923a29961631629221afa54f592c71 Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jollamobile.com>
Date: Tue, 7 Nov 2017 13:17:00 +0200
Subject: [PATCH 37/38] PK_FILTER_ENUM_ARCH / _NOT_ARCH to filter based on
 compatibility

Upstream commit 310c944e262ba3e2 changed this to filter based on
primary arch, but
a) it's wasn't taking into use the configured arch,
b) pkcon install -> pk_console_install_packages() uses enum_arch
by default presumably to get compatible packages for which there are
no better filters either.

Just assuming pkcon knows better so it's possible to pkcon install
i586 package for x86_64 device with i486 runtime. Like Jolla tablet.
---
 backends/zypp/pk-backend-zypp.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/backends/zypp/pk-backend-zypp.cpp b/backends/zypp/pk-backend-zypp.cpp
index 4815be033..13f036b31 100644
--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -1618,13 +1618,11 @@ zypp_filter_solvable (PkBitfield filters, const sat::Solvable &item)
 		if (i == PK_FILTER_ENUM_NOT_INSTALLED && item.isSystem ())
 			return TRUE;
 		if (i == PK_FILTER_ENUM_ARCH) {
-			if (item.arch () != ZConfig::defaultSystemArchitecture () &&
-			    item.arch () != "noarch")
+			if (! item.arch ().compatibleWith (ZConfig::instance().systemArchitecture ()))
 				return TRUE;
 		}
 		if (i == PK_FILTER_ENUM_NOT_ARCH) {
-			if (item.arch () == ZConfig::defaultSystemArchitecture () ||
-			    item.arch () == "noarch")
+			if (item.arch ().compatibleWith (ZConfig::instance().systemArchitecture ()))
 				return TRUE;
 		}
 		if (i == PK_FILTER_ENUM_SOURCE && !(isKind<SrcPackage>(item)))
-- 
2.13.6

